### PR TITLE
feat(#251)!: treat empty strings as missing environment variables

### PR DESCRIPTION
**Pull request by svr93**
Note: Strings containing spaces are also considered missing values.

### CODE REVIEW

**Comment by af** [comment]
Thanks, will be afk until next week but will review then!

File: src/core.ts [hunk 1]
```diff
@@ -11,16 +11,16 @@ import { defaultReporter } from './reporter'
 function validateVar<T>({
   spec,
   name,
-  rawValue,
+  normalizedValue,
```
Comment (af) [inline src/core.ts]: Let's keep the prefix `raw` here, and below in `readRawEnvValue`. Even though it's no longer strictly true (because of the trim() check), it communicates the untrusted nature of the input better IMO

File: tests/validators.test.ts [hunk 2]
```diff
@@ -39,8 +39,19 @@ test('bool() works with various formats', () => {
...
+
   const defaultF = cleanEnv({}, { FOO: bool({ default: false }) })
   expect(defaultF).toEqual({ FOO: false })
+
+  const defaultTWithWhitespace = cleanEnv(
+    { FOO: ' ' },
+    { FOO: bool({ default: true }) },
+    makeSilent,
+  )
+  expect(defaultTWithWhitespace).toEqual({ FOO: true })
```
Comment (af) [inline tests/validators.test.ts:54]: could you add a test case for a string validator w/ a `default` receiving an empty string as well?

**Review by af — COMMENTED** [review]
Implementation looks good, just a couple nits

File: src/core.ts (hunk 1 above)
Comment (svr93) [reply src/core.ts]: Fixed

File: tests/validators.test.ts (hunk 2 above)
Comment (svr93) [reply tests/validators.test.ts:54]: Fixed

**Comment by svr93** [comment]
@af thanks for review! New changes were added ⬆️

**Comment by af** [comment]
Thanks!
